pilot/pkg/security/authz/builder/extauthz.go

```diff
@@ -189,22 +189,35 @@ func parseService(in *plugin.InputParams, service string, port int) (hostname st
 		err = fmt.Errorf("service must not be empty")
 		return
 	}
-	parts := strings.Split(service, "/")
-	// TODO(yangminzhu): Currently require the user to always specify namespace, may relax this later.
-	if len(parts) != 2 {
-		err = fmt.Errorf("service not in format <namespace>/<hostname>, found: %s", service)
-		return
-	}
 
-	namespace, name := parts[0], parts[1]
-	if svc := in.Push.ServiceIndex.HostnameAndNamespace[host.Name(name)][namespace]; svc != nil {
-		// TODO(yangminzhu): Verify the service and its cluster is supported, e.g. resolution type is not OriginalDst.
-		hostname = string(svc.Hostname)
-		cluster = model.BuildSubsetKey(model.TrafficDirectionOutbound, "", svc.Hostname, port)
-		return
+	// TODO(yangminzhu): Verify the service and its cluster is supported, e.g. resolution type is not OriginalDst.
+	if parts := strings.Split(service, "/"); len(parts) == 2 {
+		namespace, name := parts[0], parts[1]
+		if svc := in.Push.ServiceIndex.HostnameAndNamespace[host.Name(name)][namespace]; svc != nil {
```

Comment:
how can this handle the single servicename without fqdn?

Reply:
We only support fqdn, short name is not supported. See https://github.com/istio/api/pull/1757 for the update to the API.